be/src/exec/vectorized/partition/partition_hash_map.h

```diff
@@ -0,0 +1,250 @@
+                if (value.chunks.empty() || value.remain_size <= 0) {
+                    if (!value.chunks.empty()) {
+                        value.chunks.back()->append_selective(*chunk, value.select_indexes.data(), 0,
+                                                              value.select_indexes.size());
+                    }
+
+                    alloc_new_buffer(value, chunk);
+                }
+                value.select_indexes.push_back(i);
+                value.remain_size--;
+            }
+
+            for (auto& [_, value] : hash_map) {
+                flush(value, chunk);
+            }
+            flush(null_key_value, chunk);
+        }
+    }
+};
+
+template <PrimitiveType primitive_type, typename HashMap>
+struct PartitionHashMapWithOneNumberKey : public PartitionHashMapBase {
+    using Iterator = typename HashMap::iterator;
+    using ColumnType = RunTimeColumnType<primitive_type>;
+    using FieldType = RunTimeCppType<primitive_type>;
+    HashMap hash_map;
+
+    PartitionHashMapWithOneNumberKey(int32_t chunk_size) : PartitionHashMapBase(chunk_size) {}
+
+    void append_chunk(ChunkPtr chunk, const Columns& key_columns) {
+        DCHECK(!key_columns[0]->is_nullable());
+        const auto* key_column = down_cast<ColumnType*>(key_columns[0].get());
+        const auto& key_column_data = key_column->get_data();
+        append_chunk_for_one_key(hash_map, chunk, [&](uint32_t offset) { return key_column_data[offset]; });
+    }
+};
+
+template <PrimitiveType primitive_type, typename HashMap>
+struct PartitionHashMapWithOneNullableNumberKey : public PartitionHashMapBase {
+    using Iterator = typename HashMap::iterator;
+    using ColumnType = RunTimeColumnType<primitive_type>;
+    using FieldType = RunTimeCppType<primitive_type>;
+    HashMap hash_map;
+    PartitionChunks null_key_value;
+
+    PartitionHashMapWithOneNullableNumberKey(int32_t chunk_size) : PartitionHashMapBase(chunk_size) {}
+
+    void append_chunk(ChunkPtr chunk, const Columns& key_columns) {
+        DCHECK(key_columns[0]->is_nullable());
+        const auto* nullable_key_column = ColumnHelper::as_raw_column<NullableColumn>(key_columns[0].get());
+        const auto& key_column_data = down_cast<ColumnType*>(nullable_key_column->data_column().get())->get_data();
+        append_chunk_for_one_nullable_key(hash_map, null_key_value, chunk, nullable_key_column,
+                                          [&](uint32_t offset) { return key_column_data[offset]; });
+    }
+};
+
+template <typename HashMap>
+struct PartitionHashMapWithOneStringKey : public PartitionHashMapBase {
+    using Iterator = typename HashMap::iterator;
+    HashMap hash_map;
+
+    PartitionHashMapWithOneStringKey(int32_t chunk_size) : PartitionHashMapBase(chunk_size) {}
+
+    void append_chunk(ChunkPtr chunk, const Columns& key_columns) {
+        DCHECK(!key_columns[0]->is_nullable());
+        const auto* key_column = down_cast<BinaryColumn*>(key_columns[0].get());
```

Comment:
what about the `LargeBinaryColumn`?

Reply:
Emm, I'm not really sure about that, but aggregate does not consider the `LargeBinaryColumn` either, maybe they can be updated together later?